packages/@aws-cdk/aws-s3-deployment/lib/lambda/index.py

```diff
@@ -117,43 +115,49 @@ def cfn_error(message=None):
 #---------------------------------------------------------------------------------------------------
 # populate all files from s3_source_zips to a destination bucket
 def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune, exclude, include):
-    # create a temporary working directory
-    workdir=tempfile.mkdtemp()
+    # create a temporary working directory in /tmp or if enabled an attached efs volume
+    if ENV_KEY_MOUNT_PATH in os.environ:
+        workdir = os.getenv(ENV_KEY_MOUNT_PATH) + "/" + str(uuid4())
+        os.mkdir(workdir)
+    else:
```

Comment:
This branch becomes unnecessary if:

- The `MOUNT_PATH` is instead passed as `TMPDIR`
- The temporary directory is cleaned up before the function returns

This isn't a must have, but there would be something gained from not having different code paths for different settings of this function...

Reply:
Even if we pass `TMPDIR` as input, we will need a similar logic in `bucket-deployment.ts` to create a `TMPDIR` in either `/tmp` or `/mnt/lambda`. 

The way i look at it is that even earlier we were creating a `TMPDIR`, its just that we have a conditional logic around it now.